### PR TITLE
(READY TO MERGE) 10 user can view all listings and browse by criteria

### PR DESCRIPTION
**Pull request by cluhring**
A user can visit the listings index and individually search listings by City, Category(House, Apt, etc), nightly rates, and guests.  I still need to write some tests before merging.  Seed data is ready for a yolo.  But I had to comment out Bryce's slug stuff on the _listings partial because that was breaking my shit.  UPDATE - Tests are passing. Bryce's namespacing stuff works.  All is well.

### CODE REVIEW

File: app/controllers/listings_controller.rb [hunk 1]
```diff
@@ -1,5 +1,14 @@
 class ListingsController < ApplicationController
   def index
-    @listings = Listing.all
+    @listings = Listing.where(nil)
```
Comment (Lydias303) [inline app/controllers/listings_controller.rb]: what is public_send?
and why are we doing @listings = Listing.where(nil) ?

File: app/controllers/listings_controller.rb [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+    filtering_params(params).each do |key, value|
+      @listings = @listings.public_send(key, value) if value.present?
```
Comment (Lydias303) [inline app/controllers/listings_controller.rb]: what is the key and the value in this case?

File: app/controllers/listings_controller.rb [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
+    end
+  end
+
+private
+
+  def filtering_params(params)
+    params.slice(:city, :max_guests, :category_id, :nightly_rate)
```
Comment (Lydias303) [inline app/controllers/listings_controller.rb]: what is params.slice? is it like params.require?

File: app/models/listing.rb [hunk 4]
```diff
@@ -9,6 +9,10 @@ class Listing < ActiveRecord::Base
   belongs_to :category
   has_many :photos
   has_many :reservations
+  scope :city, -> (city) { where city: city }
+  scope :max_guests, -> (max_guests) { where("max_guests >= ?", max_guests) }
```
Comment (Lydias303) [inline app/models/listing.rb]: like the scopes :sparkles:

File: db/seeds.rb [hunk 5]
```diff
@@ -7,197 +7,438 @@
...
+               slug: "Eve",
                }])
 
-categories = Category.create([{ name: "Denver" },
-                              { name: "New York" },
-                              { name: "Seattle" },
-                              { name: "San Francisco" },
-                              { name: "Washington"
+categories = Category.create([{ name: "House" },
+                              { name: "Apartment" },
```
Comment (Lydias303) [inline db/seeds.rb]: Im glad we used the categories in this way :sparkles:

File: db/seeds.rb [hunk 6]
```diff
@@ -7,197 +7,438 @@
...
                  description: "City so nice they named it thrice",
-                 price: 11000,
-                 image: open(path + "ex_new_york2.jpg"))
-
-item5.categories << categories[1]
-item5.save
-
-item6 = Item.new(title: "These streets will make you feel brand new",
+                 category_id: 2,
+                 max_guests: 6,
```
Comment (Lydias303) [inline db/seeds.rb]: nice job on all the seed data

File: app/controllers/listings_controller.rb (hunk 1 above)
Comment (cluhring) [inline app/controllers/listings_controller.rb]: http://apidock.com/ruby/Object/public_send

File: app/controllers/listings_controller.rb (hunk 2 above)
Comment (cluhring) [inline app/controllers/listings_controller.rb]: if you search for a city, then the key is city and the value is your "search city."  Public send then uses the key as a public method call on listings to search for your "search city.

File: app/controllers/listings_controller.rb (hunk 3 above)
Comment (cluhring) [inline app/controllers/listings_controller.rb]: slice(*keys) public
Slice a hash to include only the given keys. This is useful for limiting an options hash to valid keys before passing to a method:

File: app/models/listing.rb (hunk 4 above)
Comment (cluhring) [inline app/models/listing.rb]: Thanks! I still want to refine them so you can search by all criteria at the same time though.

File: app/controllers/listings_controller.rb (hunk 2 above)
Comment (cluhring) [inline app/controllers/listings_controller.rb]: Also, I totally just copied this from this dude's blog: http://www.justinweiss.com/blog/2014/02/17/search-and-filter-rails-models-without-bloating-your-controller/

**Comment by bryceholcomb** [comment]
Looks good Chris, thanks for knocking this out

**Pull request merged**